rustls-mbedcrypto-provider/src/kx.rs

```diff
@@ -144,17 +190,19 @@ pub static ALL_KX_GROUPS: &[&dyn SupportedKxGroup] = &[
 
 /// An in-progress ECDH key exchange.  This has the algorithm,
 /// our private key, and our public key.
-struct KeyExchange {
+struct KeyExchange<T: RngCallback> {
     name: NamedGroup,
     /// The corresponding [`agreement::Algorithm`]
     agreement_algorithm: &'static agreement::Algorithm,
     /// Private key
     priv_key: PkMbed,
     /// Public key in binary format [`EcPoint`] without compression
     pub_key: OnceLock<Vec<u8>>,
+    /// Function to provider a RNG
```

Comment:
Can we change the doc comment here too? `Callback to produce RNGs when needed`

Reply:
sure!

Reply:
done!